### ISSUE TITLE
Add generate thumbnails(preview images) button to the scanner page

### ISSUE THREAD

**Issue by assmt7** [0x90d/videoduplicatefinder#383]
**Environment**
 - OS: windows

Add a button to generate preview images on demand for current list of duplicates on the scanner page.

Context:
When decreasing the percent to find more matches, VDF places many non matching files into one super group with thousands of files that seems to almost double every 2 percent lowered. So I mark that group as not a match but VDF starts generating images for thousands of  things I know are likely not a match. I can get around this by unchecking "generate preview images", scan, mark super group as not a match, recheck "generate preview images", scan again so I can see the previews. 

Just for the record it only really becomes a problem below around 85% with 12 thumbnails.

Also I think I saw some people having problems seeing the preview Images as small, I was wondering if it is reasonable or possible to have it generate less preview images than thumbnails. For example I have 12 thumbnails set for better scanner matching purposes, but for me to verify the results, the preview images only need maybe 6 or 8 thumbnails.

But this may just be me, just a suggestion. Thank you for your work.

**Comment by 0x90d** [comment]
Instead of adding such a button ideally VDF generates thumbnails on demand, i.e. when items become visible (as an opt-in option). This would also allow higher resolution thumbnails regardless of how many scan results you have. Would that solve your problem as well?

> For example I have 12 thumbnails set for better scanner matching purposes, but for me to verify the results, the preview images only need maybe 6 or 8 thumbnails.

What if the 6 or 8 thumbnails are absolutely identical but the 6 or 4 others don't? You wouldn't be able to see.

**Comment by assmt7** [comment]
> Instead of adding such a button ideally VDF generates thumbnails on demand, i.e. when items become visible (as an opt-in option). This would also allow higher resolution thumbnails regardless of how many scan results you have. Would that solve your problem as well?

Yes, that solution would seem much better. If that is what you are considering that seems great, I was just suggesting something that might be not too much work.

> What if the 6 or 8 thumbnails are absolutely identical but the 6 or 4 others don't? You wouldn't be able to see.

I mainly use the thumbnails(along with file metadata) to visually filter out obvious wrong matches, for close visual matches I usually open the source. I was only thinking along this line in terms of saving system resources and on screen space of the app.

Thanks again
